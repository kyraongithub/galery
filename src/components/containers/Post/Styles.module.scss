.content {
  position: relative;
  margin: auto;
  width: 300px;
  margin-bottom: 40px;
  .header {
    cursor: pointer;
    div,
    img {
      padding: 0 !important;
    }
  }
  p {
    margin: 0;
    max-width: 300px;
    word-wrap: break-word;
  }
  .post {
    margin-top: 20px;
    @media only screen and (max-width: 768px) {
      width: 300px;
    }
  }
}
