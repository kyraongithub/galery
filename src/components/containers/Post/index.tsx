/* eslint-disable jsx-a11y/alt-text */
import moment from "moment";
import { Col, Image, Row } from "react-bootstrap";
import styles from "./Container.module.scss";

const Post = (props: any) => {
  const { item } = props;
  return (
    <>
      {item?.link[1]["$"].href.includes(".jpg") ||
      item?.link[1]["$"].href.includes(".png") ? (
        <div className={styles.content}>
          <Row
            className={styles.header}
            onClick={() => window.open(item?.author[0].uri[0])}
            role={"button"}
          >
            <Col xl={3} xs={3} md={3} lg={3} sm={3}>
              <Image
                thumbnail
                roundedCircle
                src={item?.author[0]["flickr:buddyicon"]}
              />
            </Col>
            <Col>
              <p>
                <b>{item?.author[0].name}</b>
              </p>
              <p>{moment(item?.published[0]).fromNow()}</p>
            </Col>
          </Row>
          <Row>
            {/* <Image
              className={styles.post}
              width="400px"
              src={item?.link[1]["$"].href}
            /> */}
            <div
              className={styles.post}
              dangerouslySetInnerHTML={{
                __html: item?.content[0]["_"].replace(
                  /^(.*?)(posted a (photo|video):)/,
                  ""
                ),
              }}
            ></div>
          </Row>
          <Row>
            {item?.category.map((category: any, index: number) => {
              return category["$"].term !== "" ? (
                <p key={index}>#{category["$"].term} </p>
              ) : null;
            })}
          </Row>
        </div>
      ) : null}
    </>
  );
};

export default Post;
