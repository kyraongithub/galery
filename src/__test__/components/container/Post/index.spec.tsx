import Post from "@/components/containers/Post";
import { fireEvent, render } from "@testing-library/react";

describe("Post component", () => {
  it("render Post", () => {
    const { container } = render(<Post />);
    expect(container).toMatchSnapshot();
  });
  it("render Post with data", () => {
    const mockData = {
      title: ["Two houses behind the fence"],
      link: [
        {
          $: {
            rel: "alternate",
            type: "text/html",
            href: "https://www.flickr.com/photos/190550837@N04/52782611707/",
          },
        },
        {
          $: {
            rel: "enclosure",
            type: "image/jpeg",
            href: "https://live.staticflickr.com/65535/52782611707_d429f8b443_b.jpg",
          },
        },
      ],
      id: ["tag:flickr.com,2005:/photo/52782611707"],
      published: ["2023-03-31T11:32:10Z"],
      updated: ["2023-03-31T11:32:10Z"],
      "flickr:date_taken": ["2023-03-31T12:29:05-08:00"],
      "dc:date.Taken": ["2023-03-31T12:29:05-08:00"],
      content: [
        {
          _: '\t\t\t<p><a href="https://www.flickr.com/people/190550837@N04/">.ilona.</a> posted a photo:</p>\n\t\n<p><a href="https://www.flickr.com/photos/190550837@N04/52782611707/" title="Two houses behind the fence"><img src="https://live.staticflickr.com/65535/52782611707_d429f8b443_m.jpg" width="240" height="135" alt="Two houses behind the fence" /></a></p>\n\n\n',
          $: {
            type: "html",
          },
        },
      ],
      author: [
        {
          name: [".ilona."],
          uri: ["https://www.flickr.com/people/190550837@N04/"],
          "flickr:nsid": ["190550837@N04"],
          "flickr:buddyicon": [
            "https://farm66.staticflickr.com/65535/buddyicons/190550837@N04.jpg?1665867420#190550837@N04",
          ],
        },
      ],
      category: [
        {
          $: {
            term: "tag",
            scheme: "https://www.flickr.com/photos/tags/",
          },
        },
      ],
      displaycategories: ["\n            "],
    };
    const { container } = render(<Post item={mockData} />);
    expect(container).toMatchSnapshot();
  });
  it("should handle click profile", () => {
    const mockData = {
      title: ["Two houses behind the fence"],
      link: [
        {
          $: {
            rel: "alternate",
            type: "text/html",
            href: "https://www.flickr.com/photos/190550837@N04/52782611707/",
          },
        },
        {
          $: {
            rel: "enclosure",
            type: "image/jpeg",
            href: "https://live.staticflickr.com/65535/52782611707_d429f8b443_b.png",
          },
        },
      ],
      id: ["tag:flickr.com,2005:/photo/52782611707"],
      published: ["2023-03-31T11:32:10Z"],
      updated: ["2023-03-31T11:32:10Z"],
      "flickr:date_taken": ["2023-03-31T12:29:05-08:00"],
      "dc:date.Taken": ["2023-03-31T12:29:05-08:00"],
      content: [
        {
          _: '\t\t\t<p><a href="https://www.flickr.com/people/190550837@N04/">.ilona.</a> posted a photo:</p>\n\t\n<p><a href="https://www.flickr.com/photos/190550837@N04/52782611707/" title="Two houses behind the fence"><img src="https://live.staticflickr.com/65535/52782611707_d429f8b443_m.jpg" width="240" height="135" alt="Two houses behind the fence" /></a></p>\n\n\n',
          $: {
            type: "html",
          },
        },
      ],
      author: [
        {
          name: [".ilona."],
          uri: ["https://www.flickr.com/people/190550837@N04/"],
          "flickr:nsid": ["190550837@N04"],
          "flickr:buddyicon": [
            "https://farm66.staticflickr.com/65535/buddyicons/190550837@N04.jpg?1665867420#190550837@N04",
          ],
        },
      ],
      category: [
        {
          $: {
            term: "",
            scheme: "https://www.flickr.com/photos/tags/",
          },
        },
      ],
      displaycategories: ["\n            "],
    };
    const { container, getByRole } = render(<Post item={mockData} />);
    fireEvent.click(getByRole("button"));
    expect(container).toMatchSnapshot();
  });
});
