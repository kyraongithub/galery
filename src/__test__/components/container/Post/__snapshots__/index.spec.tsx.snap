// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component render Post 1`] = `<div />`;

exports[`Post component render Post with data 1`] = `
<div>
  <div
    class="content"
  >
    <div
      class="header row"
      role="button"
    >
      <div
        class="col-xl-3 col-lg-3 col-md-3 col-sm-3 col-3"
      >
        <img
          class="rounded-circle img-thumbnail"
          src="https://farm66.staticflickr.com/65535/buddyicons/190550837@N04.jpg?1665867420#190550837@N04"
        />
      </div>
      <div
        class="col"
      >
        <p>
          <b>
            .ilona.
          </b>
        </p>
        <p>
          an hour ago
        </p>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="post"
      >
        <p />
        
	

        <p>
          <a
            href="https://www.flickr.com/photos/190550837@N04/52782611707/"
            title="Two houses behind the fence"
          >
            <img
              alt="Two houses behind the fence"
              height="135"
              src="https://live.staticflickr.com/65535/52782611707_d429f8b443_m.jpg"
              width="240"
            />
          </a>
        </p>
        



      </div>
    </div>
    <div
      class="row"
    >
      <p>
        #
        tag
         
      </p>
    </div>
  </div>
</div>
`;

exports[`Post component should handle click profile 1`] = `
<div>
  <div
    class="content"
  >
    <div
      class="header row"
      role="button"
    >
      <div
        class="col-xl-3 col-lg-3 col-md-3 col-sm-3 col-3"
      >
        <img
          class="rounded-circle img-thumbnail"
          src="https://farm66.staticflickr.com/65535/buddyicons/190550837@N04.jpg?1665867420#190550837@N04"
        />
      </div>
      <div
        class="col"
      >
        <p>
          <b>
            .ilona.
          </b>
        </p>
        <p>
          an hour ago
        </p>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="post"
      >
        <p />
        
	

        <p>
          <a
            href="https://www.flickr.com/photos/190550837@N04/52782611707/"
            title="Two houses behind the fence"
          >
            <img
              alt="Two houses behind the fence"
              height="135"
              src="https://live.staticflickr.com/65535/52782611707_d429f8b443_m.jpg"
              width="240"
            />
          </a>
        </p>
        



      </div>
    </div>
    <div
      class="row"
    />
  </div>
</div>
`;
